syntax = "proto3";

package paperpi;

enum RequestType {
  UNKNOWN_REQ_TYPE = 0;
  CITATION = 1;
  REFERENCE = 2;
}

message RequestConfig {
  RequestType request_type = 1;
  int32 max_pages = 2;
  int32 start_page = 3;
  int32 limit = 4;

  string links_key_in_response = 11;
  string output_dir = 12;
}

message SemanticScholarConfig {
  string seed_file = 1;

  repeated RequestConfig request_configs = 11;
}

message CrawlerConfig {
  oneof site_crawler_onfig {
    SemanticScholarConfig semantic_scholar_config = 1;
  }

  string local_cache_root = 11;
}
